server/commitlog/commitlog.go

```diff
@@ -432,7 +493,7 @@ func (l *CommitLog) Truncate(offset int64) error {
 	atomic.StorePointer((*unsafe.Pointer)(unsafe.Pointer(&l.vActiveSegment)),
 		unsafe.Pointer(activeSegment))
 	l.segments = segments
-	return nil
+	return l.leaderEpochCache.ClearLatest(offset)
```

Comment:
Add a test to ensure `Truncate` is also correctly truncating the leader epoch cache.